### PR TITLE
don't disable menu items during demo playback

### LINKED ISSUE #1439: DOOM Compatibility settings bugged
The DOOM compatibility settings are bugged, sometimes they activate after a demo ends like they're supposed to. Sometimes they don't. 
Also, could you please add an option to disable demos along with this bug fix? It's tedious waiting for it to end.

### PR DESCRIPTION
**Pull request by rfomin**
* Show disabled items immediately when complevel/strict mode is changed

* Remove S_CRITICAL

Fix #1439

### CODE REVIEW

File: src/m_menu.c [hunk 1]
```diff
@@ -3657,21 +3653,21 @@ setup_menu_t comp_settings1[] =  // Compatibility Settings screen #1
...
 
-  {"Pistol Start", S_YESNO|S_STRICT|S_CRITICAL, m_null, M_X, M_SPC,
+  {"Pistol Start", S_YESNO|S_STRICT, m_null, M_X, M_SPC,
    {"pistolstart"}},
 
   {"", S_SKIP, m_null, M_X, M_SPC},
 
-  {"Improved Hit Detection", S_YESNO|S_STRICT|S_CRITICAL, m_null, M_X, M_SPC,
-   {"blockmapfix"}},
+  {"Improved Hit Detection", S_YESNO|S_STRICT|S_BOOM, m_null, M_X,
```
Comment (fabiangreffrath) [inline src/m_menu.c:3667]: Why is this a Boom feature?

Comment (rfomin) [reply src/m_menu.c:3667]: It's Boom or higher, it won't work in vanilla complevel (conflict with `emu_intercepts`).

**Comment by fabiangreffrath** [comment]
Okay, maybe i just don't get it. But if we allow changing critical settings while a demo is running, why pause the demo at all?

**Comment by rfomin** [comment]
> Okay, maybe i just don't get it. But if we allow changing critical settings while a demo is running, why pause the demo at all?

We allow it, but the critical options won't work while playing the demo (or netgame), I think that's intuitive.

**Comment by rfomin** [comment]
> why pause the demo at all?

On second thought, maybe we don't need to pause the demo.

**Comment by fabiangreffrath** [comment]
I'll have to check out out. If it suggests "you can change this value, but not in-game and thus it's paused", then it's fine.

**Comment by rfomin** [comment]
> If it suggests "you can change this value, but not in-game and thus it's paused", then it's fine.

There is no new help message. I think we've gone overboard with "disabling items" in the past, it's obvious to most users that these options have no effect on the current demo playback.

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**